web/src/input_pill.ts

```diff
@@ -374,9 +419,9 @@ export function create(opts) {
         store.$parent.on("copy", ".pill", (e) => {
             const $element = store.$parent.find(":focus");
             const data = funcs.getByElement($element[0]);
-            e.originalEvent.clipboardData.setData(
+            (e.originalEvent as ClipboardEvent).clipboardData?.setData(
                 "text/plain",
-                store.get_text_from_item(data.item),
+                data ? store.get_text_from_item(data.item) : "",
```

Comment:
This function is a bit puzzling, but I feel like we might prefer to assert that `data` is actually defined. This function is a bit hard to trace, but I feel like there's likely something quite wrong with the setup if it fails.

What I'm pretty puzzled about is why `e` is mostly ignored by this function.

Reply:
Yeah, It actually makes sense to me to just use `e` to get the current element than to call `find(":focus")` and yeah I think we can probably do a non-null assertion here for `data` since I think it will always be defined because this callback is only running when we copy a pill which is only possible if we have it in our `store`.

This works as expected: -
```typescript
        store.$parent.on("copy", ".pill", (e) => {
            const $element = e.currentTarget as HTMLElement;
            const data = funcs.getByElement($element)!;
            (e.originalEvent as ClipboardEvent).clipboardData?.setData(
                "text/plain",
                store.get_text_from_item(data.item),
            );
            e.preventDefault();
        });
  ```